### PR TITLE
merge bitcoin #16953 #16918 #16917 #16898 #14696 #16888 #16737 #16404 #15687 #16294 : Backport

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

File: build-aux/m4/ax_boost_base.m4 [hunk 1]
```diff
@@ -127,8 +127,14 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
       [multiarch_libsubdir="lib/${host_cpu}-${host_os}"]
     )
 
+    dnl some arches may advertise a cpu type that doesn't line up with their
+    dnl prefix's cpu type. For example, uname may report armv7l while libs are
+    dnl installed to /usr/lib/arm-linux-gnueabihf. Try getting the compiler's
+    dnl value for an extra chance of finding the correct path.
+    libsubdirs="lib/`$CXX -dumpmachine 2>/dev/null` $libsubdirs"
```
Comment (UdjinM6) [inline build-aux/m4/ax_boost_base.m4:134]: 71235217d919519422af2747d205f247bff3e4e0: I think this patch was never merged into master and the issue was fixed via 18558 (which we backported already), see 18501

File: test/functional/interface_zmq.py [hunk 2]
```diff
@@ -13,10 +13,9 @@
 from test_framework.util import (
     assert_equal,
     hash256,
+    connect_nodes,
```
Comment (UdjinM6) [inline test/functional/interface_zmq.py:16]: 16404: sort

File: test/functional/p2p_connect_to_devnet.py [hunk 3]
```diff
@@ -18,13 +18,13 @@ def setup_network(self):
...
         self.start_node(1)
-        connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes(self.nodes[0], 1)
         self.sync_all()
 
 
     def run_test(self):
         self.nodes[0].add_p2p_connection(P2PInterface())
-        assert_equal(self.nodes[0].getconnectioncount(), 3)  # 2 in/out dashd + 1 p2p
+        assert_equal(self.nodes[0].getconnectioncount(), 2)  # 2 in/out dashd + 1 p2p
```
Comment (UdjinM6) [inline test/functional/p2p_connect_to_devnet.py:27]: 16898: should also fix comments (should be `1 out dashd + 1 in p2p`)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: build-aux/m4/ax_boost_base.m4 (hunk 1 above)
Comment (vijaydasmp) [reply build-aux/m4/ax_boost_base.m4:134]: ok, I will drop this commit

File: src/qt/test/paymentservertests.cpp [hunk 4]
```diff
@@ -66,7 +66,7 @@ static SendCoinsRecipient handleRequest(PaymentServer* server, std::vector<unsig
 
 void PaymentServerTests::paymentServerTests()
 {
-    SelectParams(CBaseChainParams::MAIN);
+    BasicTestingSetup testing_setup(CBaseChainParams::MAIN);
```
Comment (vijaydasmp) [inline src/qt/test/paymentservertests.cpp:69]: maintained to look like upstream

File: test/functional/p2p_invalid_messages.py [hunk 5]
```diff
@@ -81,7 +81,7 @@ def run_test(self):
 
             # Peer 1, despite serving up a bunch of nonsense, should still be connected.
             self.log.info("Waiting for node to drop junk messages.")
-            node.p2p.sync_with_ping(timeout=30)
+            node.p2p.sync_with_ping(timeout=320)
```
Comment (vijaydasmp) [inline test/functional/p2p_invalid_messages.py:84]: In upstream changes, the original timeout was 120 and changed to 320,  Is 320 ok here or shall we choose a lower

**Comment by vijaydasmp** [comment]
zmq.error.Again: Resource temporarily unavailable , Please retrigger this , looks like a temp failure

**Comment by vijaydasmp** [comment]
Seems to be a flaky test failure
https://gitlab.com/dashpay/dash/-/pipelines/428831494

File: test/functional/interface_zmq.py (hunk 2 above)
Comment (vijaydasmp) [reply test/functional/interface_zmq.py:15]: Have Updated this file , not sure why this is not marked as "Outdated"

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 requesting review

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**